src/eduid_userdb/user.py

```diff
@@ -142,15 +144,16 @@ def _parse_mail_addresses(self):
             # old-style userdb primary e-mail address indicator
             for idx in xrange(len(_mail_addresses)):
                 if _mail_addresses[idx]['email'] == self._data_in['mail']:
-                    _mail_addresses[idx]['primary'] = True
                     if 'passwords' in self._data_in:
                         # Work around a bug where one could signup, not follow the link in the e-mail
                         # and then do a password request to set a password. The e-mail address is
                         # implicitly verified by the password reset (which must have been done using e-mail).
                         _mail_addresses[idx]['verified'] = True
+                    if 'verified' in _mail_addresses[idx] and _mail_addresses[idx]['verified']:
+                        _mail_addresses[idx]['primary'] = True
             self._data_in.pop('mail')
 
-        if len(_mail_addresses) == 1:
+        if len(_mail_addresses) == 1 and 'verified' in _mail_addresses[0] and _mail_addresses[0]['verified']:
```

Comment:
"and _mail_addressses[0].get('verified', False)" is shorter and therefor easier to read IMO, but even better might be to initialize 'verified' to False if it is not present